packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -90,13 +80,6 @@ function BlockSelectionButton( { clientId, rootClientId, blockElement } ) {
 	// Focus the breadcrumb in navigation mode.
 	useEffect( () => {
 		ref.current.focus();
-
-		// NVDA on windows suffers from a bug where focus changes are not announced properly
-		// See WordPress/gutenberg#24121 and nvaccess/nvda#5825 for more details
-		// To solve it we announce the focus change manually.
-		if ( isWindows() ) {
-			speak( label );
-		}
```

Comment:
I'm really curious why this has been removed and replaced with the `aria-live` attribute, they are supposed to have the same result. I think you mentioned that `speak` was `undefined` for you, is that right? This would mean that there's probably an issue elsewhere that could have larger impacts.

Reply:
@youknowriad Yes, whenever I did a

`console.log( speak( label ) );`

it came back undefined 90% of the time when it was supposed to be called. I have a feeling this is related to how the React hook is used in combination with useEffect(). If I understand it right, useEffect() will run directly after the component is rendered so if that's the case, speak() may not be callable at this point. Instead, I just added the attribute directly to the button because the dynamic change that occurs with Tab key triggers the aria-live region to announce to screen readers. This is an easy fix in comparison to all the debugging that may have to be done to figure out why wordpress/a11y.speak() is having trouble in this context. It's an easier solution so I think it's fine to go with it.


https://www.freecodecamp.org/news/beginners-guide-to-using-react-hooks/

Reply:
I also tried to create a timeout function there to see if waiting some time would allow me access to speak() more reliably, but no dice on that either.

Reply:
`speak` function doesn't return anything meaning calling `console.log( speak( label ) );` will always print `undefined` in the console, but this doesn't mean the function didn't work. `speak` works by changing the content of a DOM element with the following class `a11y-speak-polite` at the end of the DOM

Reply:
@youknowriad No, I am certain it didn't work. The only time the speak() function seems to be fired is win tabbing from the title field to the first block or from the post options sidebar after the block list, Shift+Tab. Other than that, just tabbing through the list does not trigger the region. It does not announce or change, you can view this with the web inspector.

Also, I tried changing the 2nd arg to "assertive" and still no luck.

`speak( label, 'assertive' );`

I was just trying with console.log() to see if I could find the issue but it lead to nothing. I am still convinced it may have something to do with useEffect().

All I can say for certain is the speak() function only fires sometimes, it doesn't work all the time. E.g. if you have a list of 5 blocks, it will fire on block 1 and block 10 (Shift+Tab) but nowhere else during the list.

Reply:
Fixed.